### PR TITLE
better readability and less errors with namedtuples, fixes #823

### LINKED ISSUE #823: use namedtuples
e.g. for the chunk index values, to get rid of that ugly c[0], [1], [2].

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: borg/cache.py [hunk 1]
```diff
@@ -375,12 +378,12 @@ def add_chunk(self, id, data, stats, overwrite=False):
         data = self.key.encrypt(data)
         csize = len(data)
         self.repository.put(id, data, wait=False)
-        self.chunks[id] = (refcount + 1, size, csize)
+        self.chunks[id] = ChunkIndexEntry(refcount + 1, size, csize)
```
Comment (enkore) [inline borg/cache.py]: Use `self.chunks.add(id, 1, size, csize)` instead

Comment (ThomasWaldmann) [inline borg/cache.py]: chunks.add does not set size and csize.
if we force overwrite, we likely have a reason, e.g. changed compression or encryption.
so the same id chunk may have different csize.

Comment (enkore) [inline borg/cache.py]: Change chunks.add to use size/csize from the supplied values and not what's already in the hashindex?

Comment (ThomasWaldmann) [inline borg/cache.py]: yes, merging this now.

**Pull request merged**